packages/adapter-d1/src/d1.ts

```diff
@@ -149,6 +111,34 @@ class D1Queryable<ClientT extends StdClient> implements Queryable {
       })
     }
   }
+
+  cleanArg(arg: unknown): unknown {
+    // * Hack for booleans, we must convert them to 0/1.
+    // * ✘ [ERROR] Error in performIO: Error: D1_TYPE_ERROR: Type 'boolean' not supported for value 'true'
+    if (arg === true) {
+      return 1
+    }
+
+    if (arg === false) {
+      return 0
+    }
+
+    if (arg instanceof Uint8Array) {
```

Comment:
```suggestion
    if (ArrayBuffer.isView(arg)) {
```

Reply:
I reverted this for now because I encountered this error
```
Error: packages/adapter-d1 build: src/d1.ts(127,25): error TS2769: No overload matches this call.
packages/adapter-d1 build:   Overload 1 of 4, '(iterable: Iterable<unknown> | ArrayLike<unknown>): unknown[]', gave the following error.
packages/adapter-d1 build:     Argument of type 'ArrayBufferView' is not assignable to parameter of type 'Iterable<unknown> | ArrayLike<unknown>'.
packages/adapter-d1 build:   Overload 2 of 4, '(arrayLike: ArrayLike<unknown>): unknown[]', gave the following error.
packages/adapter-d1 build:     Argument of type 'ArrayBufferView' is not assignable to parameter of type 'ArrayLike<unknown>'.
packages/adapter-d1 build:       Property 'length' is missing in type 'ArrayBufferView' but required in type 'ArrayLike<unknown>'.
Error: s/adapter-d1 build: ✘ [ERROR] Command failed with exit code 2: tsc --project tsconfig.build.json --paths null [plugin tscPlugin]
```
Which I first tried to solve by changing
```
if (ArrayBuffer.isView(arg)) {
      return Array.from(arg)
}
```
to
```js
if (ArrayBuffer.isView(arg)) {
    return Buffer.from(arg.buffer, arg.byteOffset, arg.byteLength)
}
```

Which in turn resulted in errors like 
```
 FAIL  tests/functional/raw-queries/send-type-hints/tests.ts
  ● Console

    console.error
      Error in performIO: Error [AssertionError]: false == true
          at new AssertionError (node-internal:internal_assertionerror:419:15)
          at assert (node-internal:internal_assert:49:15)
          at ArrayBufferView (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/miniflare/src/workers/core/devalue.ts:95:3)
          at hydrate (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/.pnpm/devalue@4.3.2/node_modules/devalue/src/parse.js:60:32)
          at hydrate (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/.pnpm/devalue@4.3.2/node_modules/devalue/src/parse.js:115:17)
          at unflatten (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/.pnpm/devalue@4.3.2/node_modules/devalue/src/parse.js:132:9)
          at parse (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/.pnpm/devalue@4.3.2/node_modules/devalue/src/parse.js:16:9)
          at parseWithReadableStreams (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/miniflare/src/workers/core/devalue.ts:313:9)
          at #fetch (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/miniflare/src/workers/core/proxy.worker.ts:237:12)
          at async ProxyServer.fetch (file:///Users/j42/Dev/prisma/node_modules/.pnpm/miniflare@3.20240129.2/node_modules/miniflare/src/workers/core/proxy.worker.ts:125:11) {
        [cause]: undefined
      }
```